### ISSUE TITLE
cookiecutter .taskcluster.yml for github doesn't work for pushes to non-main branches

### ISSUE THREAD

**Issue by bhearsum** [taskcluster/taskgraph#242]
The cookiecutter file sets `base_ref` and `base_rev`. This puts us into [this block of run-task](https://github.com/taskcluster/taskgraph/blob/a0dd5d37b31b4912969b87bd57c14462cb0a1868/src/taskgraph/run-task/run-task#L635-L644) which ultimately runs a command like: ```git fetch origin refs/heads/dev-enable-more-branches``` - which fails, because only the main branch of a repository has an entry in `refs/heads`. Anywhere that I've found that has pushes to non-main branches ends up omitting these parameters altogether, eg: [Mozilla VPN](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/releases/2.14.1/.taskcluster.yml#L261-L266).

Full log of a failed decision task included for completeness:
```
[vcs 2023-05-15T19:57:56.490Z] executing ['git', 'clone', 'https://github.com/mozilla-releng/staging-firefox-translations-training', '/builds/worker/checkouts/src']
[vcs 2023-05-15T19:57:56.493Z] Cloning into '/builds/worker/checkouts/src'...
[vcs 2023-05-15T19:57:56.886Z] executing ['git', 'fetch', 'origin', 'refs/heads/dev-enable-more-branches']
[vcs 2023-05-15T19:57:57.031Z] From https://github.com/mozilla-releng/staging-firefox-translations-training
[vcs 2023-05-15T19:57:57.031Z]  * branch            dev-enable-more-branches -> FETCH_HEAD
[vcs 2023-05-15T19:57:57.035Z] executing ['git', 'checkout', 'refs/heads/dev-enable-more-branches']
[vcs 2023-05-15T19:57:57.038Z] error: pathspec 'refs/heads/dev-enable-more-branches' did not match any file(s) known to git
```

**Comment by jcristau** [comment]
Is the issue here that we should be passing `--base-ref=dev-enabled-more-branches` instead of `--base-ref=refs/heads/dev-enable-more-branches`?

**Comment by bhearsum** [comment]
Yeah, that's another possible solution. `run-task` might be able to cope with the scenario better as well. (I haven't given this too much thought - just wanted to get it on file.)
